library/general/requiredannotations/template.yaml

```diff
@@ -18,13 +19,21 @@ spec:
               type: string
             annotations:
               type: array
+              description: >-
+                A list of annotations and values the object must specify.
               items:
                 type: object
                 properties:
                   key:
                     type: string
+                    description: >-
+                      The required annotation.
                   allowedRegex:
                     type: string
+                    description: >-
+                      A regular expression the annotation's value must match. The value
+                      must contain at least one match for the regular expression.
+                      If unset, or set to empty string, matches all values.
```

Comment:
nit: I think we can ignore the empty-string case... is an empty string regex one that matches everything?

Reply:
Sort of - the wording is weird. The empty string regex has matches in all existent strings. It doesn't "match" the text (like ".*" would), but it "has matches" in the text.

Reply:
That sounds like it's covered by "must contain at least one match for the regular expression"

For the undefined case, you can prefix the first sentence with "When defined, a regular expression..."

Basically, it'd be good to make it clear that users can ignore this field unless they want to leverage it.

Reply:
Done.